source/common/router/upstream_request.cc

```diff
@@ -781,6 +797,33 @@ void UpstreamRequest::clearRequestEncoder() {
   upstream_.reset();
 }
 
+void UpstreamRequest::readDisableOrDefer(bool disable) {
+  if (disable) {
+    if (parent_.downstreamResponseStarted()) {
```

Comment:
it would be good to have the behavior here clearly commented.
I'd comment here, see comments on deferred_read_disabling_count_ for when we do and don't defer, and the comment on the member variable why we do this (optionally  even link to the issue)